From 552435a9e0ad9db7e5049012c3d194144d86a093 Mon Sep 17 00:00:00 2001
From: Chen Baozi <chenbaozi@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:04 +0800
Subject: [PATCH 050/150] dt-bindings: ipmi: Add bindings for Phytium BT

This patch documents the DT bindings for Phytium BT interface.

Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Change-Id: I44a6032d92154c6b647c68d20c16eb811a9517b1
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 .../bindings/ipmi/phytium,bt-bmc.yaml         | 39 +++++++++++++++++++
 MAINTAINERS                                   |  1 +
 2 files changed, 40 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/ipmi/phytium,bt-bmc.yaml

diff --git a/Documentation/devicetree/bindings/ipmi/phytium,bt-bmc.yaml b/Documentation/devicetree/bindings/ipmi/phytium,bt-bmc.yaml
new file mode 100644
index 000000000000..8b62bb0eddfa
--- /dev/null
+++ b/Documentation/devicetree/bindings/ipmi/phytium,bt-bmc.yaml
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/ipmi/phytium,bt-bmc.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Phytium BT (Block Transfer) IPMI interface
+
+maintainers:
+  - Chen Baozi <chenbaozi@phytium.com.cn>
+
+description: |
+  The Phytium E-series SOCs can be used in BMC which may have a BT
+  interface used to perform in-band IPMI communication with their host.
+
+properties:
+  compatible:
+    const: phytium,bt-bmc
+
+  interrupts:
+    maxItems: 1
+
+  reg:
+    maxItems: 1
+
+required:
+  - compatible
+  - interrupts
+  - reg
+
+additionalProperties: false
+
+examples:
+  - |
+    bt: bt@250 {
+      compatible = "phytium,bt-bmc";
+      reg = <0x250 0x1c>;
+      interrupts = <GIC_SPI 88 IRQ_TYPE_LEVEL_HIGH>;
+    };
diff --git a/MAINTAINERS b/MAINTAINERS
index 218aee65cede..bf4fb45eb7b9 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -2537,6 +2537,7 @@ F:	Documentation/devicetree/bindings/i2c/phytium,i2c.yaml
 F:	Documentation/devicetree/bindings/media/phytium,jpeg.yaml
 F:	Documentation/devicetree/bindings/iio/adc/phytium,adc.yaml
 F:	Documentation/devicetree/bindings/input/phytium,keypad.yaml
+F:	Documentation/devicetree/bindings/ipmi/phytium,bt-bmc.yaml
 F:	Documentation/devicetree/bindings/ipmi/phytium,kcs-bmc.yaml
 F:	Documentation/devicetree/bindings/mmc/phytium,mci.yaml
 F:	Documentation/devicetree/bindings/mmc/phytium,sdci.yaml
-- 
2.47.0

